integration-tests/bats/profile.bats

```diff
@@ -0,0 +1,315 @@
+#!/usr/bin/env bats
+load $BATS_TEST_DIRNAME/helper/common.bash
+load $BATS_TEST_DIRNAME/helper/query-server-common.bash
+
+make_repo() {
+  mkdir "$1"
+  cd "$1"
+  dolt init
+  dolt sql -q "create table $1_tbl (id int)"
+  dolt sql <<SQL
+CREATE TABLE table1 (pk int PRIMARY KEY);
+CREATE TABLE table2 (pk int PRIMARY KEY);
+SQL
+  dolt add -A && dolt commit -m "tables table1, table2"
+  cd ..
+}
+
+setup() {
+    setup_no_dolt_init
+    make_repo defaultDB
+    make_repo altDB
+
+    unset DOLT_CLI_PASSWORD
+    unset DOLT_SILENCE_USER_REQ_FOR_TESTING
+}
+
+teardown() {
+    stop_sql_server 1
+    teardown_common
+}
+
+@test "profile: --profile exists and isn't empty" {
+    cd defaultDB
+    dolt sql -q "create table test (pk int primary key)"
+    dolt sql -q "insert into test values (999)"
+    dolt add test
+    dolt commit -m "insert initial value into test"
+    cd ..
+
+    dolt profile add --use-db defaultDB defaultTest
+    run dolt --profile defaultTest sql -q "select * from test"
+    [ "$status" -eq 0 ] || false
+    [[ "$output" =~ "999" ]] || false
+}
+
+@test "profile: --profile doesn't exist" {
+    dolt profile add --use-db defaultDB defaultTest
+
+    run dolt --profile nonExistentProfile sql -q "select * from altDB_tbl"
+    [ "$status" -eq 1 ] || false
+    [[ "$output" =~ "Failure to parse arguments: profile nonExistentProfile not found" ]] || false
+}
+
+@test "profile: additional flag gets used" {
+    cd altDB
+    dolt sql -q "create table test (pk int primary key)"
+    dolt sql -q "insert into test values (999)"
+    dolt add test
+    dolt commit -m "insert initial value into test"
+    cd ..
+
+    dolt profile add --user dolt --password "" userProfile
```

Comment:
To take this a little further into the experience I would expect as a user.

`dolt --user pat sql` -  will prompt for a password because none was specified.

So if you create a profile:

`dolt profile add --user pat newprofile`

then 

`dolt --profile newprofile sql` should prompt for a password.

I believe it's impossible to do this with the current mechanism. You may need additional flags in the serialized object to ensure you have all the information you need.